### PR TITLE
IDE-967 Update and improve SALT colorizer

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Signed-off-by: David de Hilster <david.dehilster@lexisnexis.com>

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: EclEditor/LexSALT.cxx [hunk 1]
```diff
@@ -255,21 +161,9 @@ static void ColouriseSaltDoc(unsigned int startPos, int length, int initStyle, W
...
-                    }
-                    else {
-                        sc.ForwardSetState(SCE_SALT_ARGUMENTS);
-                    }
-                    parens++;
-                }
-            }
+			else if (isoperator(static_cast<char>(sc.ch))) {
+				sc.SetState(SCE_ECL_OPERATOR);
+			}
```
Comment (GordonSmith) [inline EclEditor/LexSALT.cxx:166]: Minor:  Indentation (tabs should be spaces)

File: EclEditor/LexSALT.cxx [hunk 2]
```diff
@@ -179,70 +140,15 @@ static void ColouriseSaltDoc(unsigned int startPos, int length, int initStyle, W
...
-            if (lineType == LINETYPE_FIELDNAME || lineType == LINETYPE_FIELDLIST_COMMAS || lineType == LINETYPE_FIELDLIST_COMMAS)
-            {
-                sc.ChangeState(SCE_SALT_FIELDNAME);
-            }
             sc.SetState(SCE_SALT_DEFAULT);
-            lineType = LINETYPE_NONE;
-            collectingFieldName = false;
-            colonCount = 0;
-            skip = 0;
+			lineStart = true;
```
Comment (GordonSmith) [inline EclEditor/LexSALT.cxx:144]: Minor: Indentation (tabs should be spaces)

File: EclEditor/LexSALT.cxx [hunk 3]
```diff
@@ -102,57 +94,26 @@ static void ColouriseSaltDoc(unsigned int startPos, int length, int initStyle, W
...
+				if (keywords0.InList(s) && lineStart) {
+					sc.ChangeState(SCE_SALT_LINE);
+				}
+				else if (keywords1.InList(s)) {
+					sc.ChangeState(SCE_SALT_ARGUMENT);
+				}
+				sc.SetState(SCE_ECL_DEFAULT);
+			}
+			break;
+		case SCE_SALT_OPERATOR:
```
Comment (GordonSmith) [inline EclEditor/LexSALT.cxx:110]: Minor: Indentation (tabs should be spaces)

File: EclEditor/LexSALT.cxx [hunk 4]
```diff
@@ -49,30 +49,22 @@ static bool IsSpaceEquiv(int state) {
...
-    WordList &keywords0 = *keywordlists[0];  // Most first parameters
-    WordList &keywords1 = *keywordlists[1];  // Common Parameters
-    WordList &keywords2 = *keywordlists[2];  // Parameters with a field
-    WordList &keywords3 = *keywordlists[3];  // Parameters with fieldlist
-    WordList &keywords4 = *keywordlists[4];  // Fieldname lists with commas
-    WordList &keywords5 = *keywordlists[5];  // Fieldnames skip one first
+    WordList &keywords0 = *keywordlists[0];  // Functions that start a line
+    WordList &keywords1 = *keywordlists[1];  // All the other arguments and keywords
+
+	CharacterSet setWord(CharacterSet::setAlphaNum, "._", 0x80, true);
```
Comment (GordonSmith) [inline EclEditor/LexSALT.cxx:62]: Minor: Indentation (tabs should be spaces)

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

**Comment by GordonSmith** [comment]
@ghalliday (it probably should be targeting 7.10.x but I am ok with it in 7.8.x as well.

**Pull request merged**